src/components/input-dropdown/input-dropdown.component.spec.ts

```diff
@@ -206,4 +212,40 @@ describe('InputDropdownComponent', () => {
         expect(openChangeSpy).toHaveBeenCalledTimes(2);
         expect(openChangeSpy).toHaveBeenCalledWith(false);
     });
+
+    it('should clear the value when allowNull is set to true', async () => {
+        component.allowNull = true;
+        component.dropdownOpen = true;
+
+        component.selected = 'One';
+        fixture.detectChanges();
+        await fixture.whenStable();
```

Comment:
There's a technique you can use to do setup ahead of running a test to avoid repeating it across multiple tests:

```ts
    describe('with allowNull = true', () => {
        beforeEach(() => {
            component.allowNull = true;
            component.dropdownOpen = true;
            component.selected = 'One';
            fixture.detectChanges();
        });
```